common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -2097,7 +2098,12 @@ def _update_item_from_fields(self, user_id, course_key, block_key, partitioned_f
 
                 # fetch and return the new item--fetching is unnecessary but a good qc step
                 new_locator = course_key.make_usage_key(block_key.type, block_key.id)
-                return self.get_item(new_locator, **kwargs)
+                new_item = self.get_item(new_locator, **kwargs)
+
+                if block_key.type == 'library_content':
```

Comment:
This is still the wrong place (or at least, not an ideal place). The code should live in the `LibraryContentBlock`, rather than having to hard-code a particular block type.